pkgs/development/tools/sourcetrail/default.nix

```
@@ -1,77 +1,29 @@
-{ stdenv, fetchurl, autoPatchelfHook
-, zlib, expat, dbus, openssl, python3 }:
+{ stdenv, fetchFromGitHub
+, boost168, cmake, qt512
+, dbus, expat, git, openssl, python3, zlib }:
 
 stdenv.mkDerivation rec {
   pname = "sourcetrail";
-  version = "2019.3.46";
+  version = "2019.4.61";
 
-  src = fetchurl {
-    name = "sourtrail.tar.gz";
-    url = "https://www.sourcetrail.com/downloads/${version}/linux/64bit";
-    sha256 = "0dnkxr3fijcsbn6pd20lyxhr9ns6ji1c4dffly9s16yq4x9ad5r4";
+  src = fetchFromGitHub {
+    owner = "CoatiSoftware";
+    repo = pname;
+    rev = version;
+    sha256 = "0srv5nnyhbxssb3ss94i3j0270rw1p0z9w2c8s96n94f5nksri7c";
   };
 
-  nativeBuildInputs = [ autoPatchelfHook ];
-  buildInputs = [ zlib expat dbus stdenv.cc.cc openssl python3 ];
+  # TODO: figure out a way to do this using cmakeFlags or something
+  patches = [ ./cmake-boost-shared-lib.patch ];
 
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p $out/bin $out/opt
-
-    mv -v setup/share $out
-    mv -v data/gui/icon/logo_1024_1024.png $out/share/icons/sourcetrail.png
-    mv -v data/gui/icon/project_256_256.png $out/share/icons/project-sourcetrail.png
-
-    mkdir -p $out/share/sourcetrail/doc
-    mv -v README EULA.txt $out/share/sourcetrail/doc
-    mv -v plugin $out/share/sourcetrail
-
-    cp -rv . $out/opt
-
-    rm $out/opt/lib/libssl.so
-    rm $out/opt/lib/platforms/{libqeglfs.so,libqwebgl.so}
-    ln -s ${openssl}/lib/libssl.so $out/opt/lib/libssl.so
-
-    substituteInPlace \
-      $out/share/applications/sourcetrail.desktop \
-      --replace /usr/bin/ $out/bin/
-
-    cat <<EOF > $out/bin/sourcetrail
-    #! ${stdenv.shell} -e
-
-    # XXX: Sourcetrail somehow copies the initial config files into the home
-    # directory without write permissions. We currently just copy them
-    # ourselves to work around this problem.
-    setup_config() {
-      local src dst
-
-      [ ! -d ~/.config/sourcetrail ] && mkdir -p ~/.config/sourcetrail
-      for src in $out/opt/data/fallback/*; do
-        dst=~/.config/sourcetrail/"\$(basename "\$src")"
-        if [ ! -e "\$dst" ]; then
-          cp -r "\$src" "\$dst"
-        fi
-      done
-
-      chmod -R u+w ~/.config/sourcetrail
-    }
-
-    [ -d "\$HOME" ] && setup_config
-    export PATH="\$PATH:${python3}/bin"
-    exec "$out/opt/Sourcetrail.sh" "\$@"
-    EOF
-
-    chmod +x $out/bin/sourcetrail
-
-    runHook postInstall
-  '';
+  nativeBuildInputs = [ boost168 cmake qt512.qtbase qt512.qtsvg ];
+  buildInputs = [ dbus expat git openssl python3 zlib ];
```

```suggestion
  buildInputs = [ boost168 qt512.qtbase qt512.qtsvg dbus expat git openssl python3 zlib ];
```

Is `git` needed at runtime? I doubt it will get picked up from here.
Are specific versions of boost and Qt needed? If so, there should be a comment with explanation. Though more generic versions are preferred whenever possible.

---

https://github.com/CoatiSoftware/Sourcetrail#git

---

Then it is likely executed at build time and should go to `nativeBuildInputs`. Though note that `fetchFromGitHub` strips the `.git` directory so `git` will not be of any help.